include/babeltrace/durin.h

```diff
@@ -0,0 +1,218 @@
+#ifndef _BABELTRACE_DURIN_H
+#define _BABELTRACE_DURIN_H
+
+/*
+ * Babeltrace - DWARF Information Reader
+ *
+ * Copyright 2015 Antoine Busque <abusque@efficios.com>
+ *
+ * Author: Antoine Busque <abusque@efficios.com>
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+#include <stdint.h>
+#include <stdlib.h>
+#include <dwarf.h>
+#include <elfutils/libdw.h>
+#include <babeltrace/babeltrace-internal.h>
+
+/*
+ * Durin is a wrapper over libdw providing a nicer, higher-level
+ * interface, to access basic debug information. Durin is also the
+ * eldest of the seven Fathers of the Dwarves created by the Vala Aulë
+ * in the lore of Tolkien's Middle-earth.
```

Comment:
To be fair, "personality" is not something I look for in a codebase.

I have no idea what _GIMPLE_ means, which is not a good thing. At least with `bt_dwarf` I immediately know "oh, it's that thing that deals with dwarf". (assuming you know what DWARF is... but at least DWARF is a well-known standard)